lang/java/avro/src/main/java/org/apache/avro/specific/SpecificDatumWriter.java

```diff
@@ -71,6 +71,21 @@ protected void writeString(Schema schema, Object datum, Encoder out)
     writeString(datum, out);
   }
 
+  @Override
+  protected void writeRecord(Schema schema, Object datum, Encoder out)
+    throws IOException {
+    if (SpecificData.get().useCustomCoders()
```

Comment:
should use this.getSpecificData() instead of the static instance

Reply:
ok